From 01cd6df17918b3a69252e4fb8744d0a2ec10aa8a Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Tue, 24 May 2016 17:07:01 -0400
Subject: [PATCH 1787/2094] drm/amdgpu: Remove unnecessary kmapping of HSA GTT
 BOs

Kmapping of all HSA GTT BOs is unnecessary and limits the maximum
size of GTT BOs that can be mapped to 4GB.

Change-Id: I2ed8eb7a4e5e2c9403c8ca2ce0530f0179db2694
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 6cf2ec2..a094aa4 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -155,13 +155,6 @@ static int try_pin_bo(struct amdgpu_bo *bo, uint32_t domain)
 		ret = amdgpu_bo_pin(bo, domain, NULL);
 		if (ret != 0)
 			goto error;
-		if (domain == AMDGPU_GEM_DOMAIN_GTT) {
-			ret = amdgpu_bo_kmap(bo, NULL);
-			if (ret != 0) {
-				pr_err("amdgpu: failed kmap GTT BO\n");
-				goto error;
-			}
-		}
 	} else {
 		/* amdgpu_bo_pin doesn't support userptr. Therefore we
 		 * can use the bo->pin_count for our version of
@@ -186,8 +179,6 @@ static int unpin_bo(struct amdgpu_bo *bo)
 {
 	int ret = 0;
 
-	amdgpu_bo_kunmap(bo);
-
 	if (!amdgpu_ttm_tt_get_usermm(bo->tbo.ttm)) {
 		ret = amdgpu_bo_unpin(bo);
 		if (ret != 0)
-- 
2.7.4

